cl/disclosures/tasks.py

```diff
@@ -216,142 +224,180 @@ def save_disclosure(extracted_data: dict, disclosure) -> None:
     :param extracted_data: disclosure
     :return:None
     """
-    addendum = "Additional Information or Explanations"
-
-    # Process and save our data into the system.
-    disclosure.has_been_extracted = True
-    disclosure.addendum_content_raw = extracted_data[addendum]["text"]
-    disclosure.addendum_redacted = extracted_data[addendum]["is_redacted"]
-    disclosure.is_amended = extracted_data.get("amended") or False
-    disclosure.report_type = get_report_type(extracted_data)
-    disclosure.save()
-
-    Investment.objects.bulk_create(
-        [
-            Investment(
-                financial_disclosure=disclosure,
-                redacted=any(v["is_redacted"] for v in investment.values()),
-                description=investment["A"]["text"],
-                page_number=investment["A"]["page_number"],
-                has_inferred_values=investment["A"]["inferred_value"],
-                income_during_reporting_period_code=investment["B1"]["text"],
-                income_during_reporting_period_type=investment["B2"]["text"],
-                gross_value_code=investment["C1"]["text"],
-                gross_value_method=investment["C2"]["text"],
-                transaction_during_reporting_period=investment["D1"]["text"],
-                transaction_date_raw=investment["D2"]["text"],
-                transaction_date=get_date(
-                    investment["D2"]["text"], disclosure.year
-                ),
-                transaction_value_code=investment["D3"]["text"],
-                transaction_gain_code=investment["D4"]["text"],
-                transaction_partner=investment["D5"]["text"],
+    try:
+        with transaction.atomic():
```

Comment:
Why move the transaction like this? I sense there's a reason I'm forgetting?

Reply:
the point was to catch and log the sentry error with more control.